test/testvm.py

```diff
@@ -664,19 +708,29 @@ def _choose_macaddr(self):
                 break
             elif not flavor:
                 macaddrs.append(macaddr)
-        for macaddr in macaddrs:
-            if self._lock_resource(macaddr):
-                return macaddr
-        raise Failure("Couldn't find unused mac address for '%s': %s" % self.flavor, resources)
-
-    def _locate_qemu_kvm(self):
-        rhel_qemu_kvm_path = '/usr/libexec/qemu-kvm'
-        if os.path.exists(rhel_qemu_kvm_path):
-            return rhel_qemu_kvm_path
-        # Assume it's in $PATH
-        return 'qemu-kvm'
-
-    def _start_qemu(self, maintain=False, tty=False, monitor=None):
+        start_time = time.time()
+        # wait a bit before we give up
+        while (time.time() - start_time) < timeout_sec:
```

Comment:
One example is check-realms: All sub-tests need a clean ipa instance. We either give them individual mac addresses or allow a bit of time here - otherwise we set ourselves up for a race. When one test finishes, the virtual machines are killed and libvirt / the os are left to clean up the mess. This includes the lock file we create for the mac address. If the new test starts before the old one is cleaned up, we get a failure. I saw this happen on the test server during previous runs.